RecommendationEngine/controllers/recoController.go

```diff
@@ -0,0 +1,76 @@
+package controllers
+
+import (
+     "encoding/json"	 
+	 "net/http"
+	 "io/ioutil"
+	 "fmt"
+     "Utopia-Profile-Mgmt/utils"
+)
+
+// Struct for converting the JSON response to obj
+type UserHistory struct {
+	Message string `json:"message"`
+	Data    []struct {
+		ID      string        `json:"_id"`
+		UserID  string        `json:"userId"`
+		History []interface{} `json:"history"`
+		Mood    []interface{} `json:"mood"`
+	} `json:"data"`
+}
+
+
+// Hits UserMgmt micro-service to fetch recent mood/valence counts.
+// Finds avg valence and returns that value to caller
+// Returns a JSON with avg value as data
+// NB: JWT verification happens at the second micro-service for this func
+func GetRecommendedValence(w http.ResponseWriter, r *http.Request)  {
+	incomingAuthHeader := r.Header.Get("Authorization")
+	
+	// Extracting token (removes "Bearer ")
+	token := incomingAuthHeader[7:]
+	
+	// Passing the JWT token along to the next micro-service...
+	request, _ := http.NewRequest("GET", "http://localhost:3001/getHistoryAndMood", nil)
```

Comment:
Remove "http://localhost:3001" from here and declare a variable, so that it will be easy to update in future.

Reply:
Done